contracts/cw721-extension/src/execute.rs

```diff
@@ -0,0 +1,149 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{DepsMut, Env, MessageInfo, Response};
+use cw2::set_contract_version;
+use cw721_base::{ContractError, InstantiateMsg};
+
+use crate::msg::ExecuteMsg;
+use crate::state::{Cw721MetadataContract, CONTRACT_NAME, CONTRACT_VERSION};
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    mut deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    let res = Cw721MetadataContract::default().instantiate(deps.branch(), env, info, msg)?;
+    // Explicitly set contract name and version, otherwise set to cw721-base info
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)
+        .map_err(ContractError::Std)?;
+    Ok(res)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    Cw721MetadataContract::default().execute(deps, env, info, msg)
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::state::{Extension, Metadata};
+
+    use super::*;
+
+    use cosmwasm_std::{
+        testing::{mock_dependencies, mock_env, mock_info},
+        DepsMut, MessageInfo,
+    };
+    use cw721::Cw721Query;
+    use cw721_base::{ExecuteMsg, MintMsg};
+
+    const CREATOR: &str = "creator";
+
+    fn instantiate_contract<'a>(deps: DepsMut, info: MessageInfo) -> Cw721MetadataContract<'a> {
+        let contract = Cw721MetadataContract::default();
+
+        // instantiate contract
+        let init_msg = InstantiateMsg {
+            name: "Ark NFT Multichain".to_string(),
+            symbol: "Ark Protocol".to_string(),
+            minter: CREATOR.to_string(),
+        };
+        contract
+            .instantiate(deps, mock_env(), info.clone(), init_msg)
+            .unwrap();
+        contract
+    }
+
+    fn mint<'a>(
+        token_id: String,
+        owner: String,
+        contract: &Cw721MetadataContract,
+        deps: DepsMut,
```

Comment:
I think its best to have deps and info as first arguments, to follow other contracts

Reply:
Ok, done.